packages/quantic/force-app/main/default/lwc/quanticTab/e2e/quanticTab.e2e.ts

```diff
@@ -0,0 +1,87 @@
+import {testSearch, testInsight, expect} from './fixture';
+import {
+  useCaseEnum,
+  useCaseTestCases,
+} from '../../../../../../playwright/utils/useCase';
+
+const fixtures = {
+  search: testSearch,
+  insight: testInsight,
+};
+
+const expectedActionCause = 'interfaceChange';
+const expectedOriginContext = 'Search';
+const expectedTabsLabels = ['All', 'Case', 'Knowledge'];
+
+useCaseTestCases.forEach((useCase) => {
+  let test = fixtures[useCase.value];
+
+  test.describe(`quantic tab ${useCase.label}`, () => {
+    test.describe('when clicking on a tab', () => {
+      test('should trigger a new search and send the correct UA analytics event', async ({
+        tab,
+        search,
+      }) => {
+        const selectedTabIndex = 0;
+        const searchResponsePromise = search.waitForSearchResponse();
+        const uaRequestPromise = tab.waitForTabSelectUaAnalytics();
+        await tab.clickTabButton(selectedTabIndex);
+
+        const searchResponse = await searchResponsePromise;
+        const searchResponseBody = searchResponse.request().postDataJSON();
+
+        expect(searchResponseBody).toEqual(
+          expect.objectContaining({
+            analytics: expect.objectContaining({
+              actionCause: expectedActionCause,
+              originContext: expectedOriginContext,
+            }),
+            tab: await tab.tabLabel(selectedTabIndex),
+          })
+        );
+
+        const uaRequest = await uaRequestPromise;
+        const uaRequestBody = uaRequest.postDataJSON();
+
+        expect(uaRequestBody).toEqual(
+          expect.objectContaining({
+            actionCause: expectedActionCause,
+            originContext: expectedOriginContext,
+            customData: expect.objectContaining({
+              interfaceChangeTo: await tab.tabLabel(selectedTabIndex),
+            }),
+          })
+        );
+      });
+    });
+
+    if (useCase.value === useCaseEnum.search) {
+      test.describe('when loading selected tab from URL', () => {
+        test('should make the right tab active', async ({tab}) => {
+          const desiredTabIndex = 1;
+          const desiredTabLabel = await tab.tabLabel(desiredTabIndex);
+          await tab.page.goto(tab.page.url() + `#tab=${desiredTabLabel}`);
```

Comment:
This should be improved, we have a similar scenario in the pager example that could serve as an example.

Basically going to the right url should be done by the fixture in the first place,

that's an example you can follow:
https://github.com/coveo/ui-kit/blob/c89e2531de170cc2396ce9981c654716d8bbfca7/packages/quantic/force-app/main/default/lwc/quanticPager/e2e/quanticPager.e2e.ts#L84-L98

https://github.com/coveo/ui-kit/blob/c89e2531de170cc2396ce9981c654716d8bbfca7/packages/quantic/force-app/main/default/lwc/quanticPager/e2e/fixture.ts#L32-L41